frontend/src/services/AuthService.ts

```diff
@@ -34,7 +34,15 @@ async function login() {
         https://docs.amplify.aws/lib/auth/social/q/platform/js/
         https://docs.amplify.aws/lib/auth/advanced/q/platform/js/#identity-pool-federation
     */
-    Auth.federatedSignIn();
```

Comment:
Not using Auth.federatedSignIn seems to bypass the Amplify library entirely, at least for login. Is there no way to pass in the identityProvider option into the signin method? Documentation suggests this is possible - e.g. 
After configuring the OAuth endpoints (Cognito Hosted UI), you can integrate your App by invoking Auth.federatedSignIn() function. Passing LoginWithAmazon, Facebook, Google, or SignInWithApple on the provider argument (e.g Auth.federatedSignIn({ provider: 'LoginWithAmazon' }))

Reply:
Updated to use customProvider option as mentioned in this example: 
https://github.com/aws-amplify/amplify-js/issues/10900